### DISCUSSION TITLE
Micropython examples?

### DISCUSSION THREAD

**Discussion by dave0003** [witnessmenow/ESP32-Cheap-Yellow-Display#238] [Q&A]
I've been able to install Micropython on my CYD, but having difficulty finding examples.  My goal is to pull a 320x240 jpg from a camera on my LAN, convert it to rgb565, and display it (and repeat).  A cheap webcam display.

There are a few examples in the rdagger repository, but I'm not sure how applicable they are for the micropython install from jtobinart.  In any case, the image display examples all save to flash first, and also use PIL to do the image conversion.  I need to save, convert, and display the image in memory.  And I haven't found any implementation of PIL for micropython, so I'm not sure where that's coming from.

Any help, pointers, examples would be appreciated.  Thanks.

**Comment by clach04** [comment]
I think this will be a tricky task with the amount of free memory available to MicroPython applications on a CYD. There isn't enough for a full screen color framebuffer.

It might be possible to do it in chunks but even then, having a web service do the conversion is probably your best bet (or switch to C).

**Reply by dave0003** [reply]
Thanks.  I ended up using this:

https://github.com/witnessmenow/ESP32-Cheap-Yellow-Display/discussions/82

I would have liked to use MicroPython, but it's ok.
